tests/src/test/scala/cilib/BenchmarkTests.scala

```diff
@@ -657,12 +732,69 @@ object BenchmarksTest extends Properties("Benchmarks") {
     trecanni((-2.0, 0.0)) === 0.0
   }
 
+  property("trigonometric1") = forAll(genNEL(0.0, pi)) { g =>
+    trigonometric1(g) >= 0.0
+  } && forAll(genConst(0.0)) { g =>
+    trigonometric1(g) === 0.0
+  }
+
+  property("trigonometric2") = forAll(genNEL(-500.0, 500.0)) { g =>
+    trigonometric2(g) >= 1.0
+  } && forAll(genConst(0.9)) { g =>
+    trigonometric2(g) === 1.0
+  }
+
+  val genUrsem1 =
+    (Gen.choose(-2.5, 3.0) |@| Gen.choose(-2.0, 2.0)) { Tuple2.apply }
+
+  property("ursem1") = forAll(genUrsem1) { g =>
+    ursem1(g) >= -4.8168
+  } && ursem1((1.69714, 0.0)) ~ (-4.8168, epsilonF(4))
+
+  val genUrsem3 =
+    (Gen.choose(-2.0, 2.0) |@| Gen.choose(-1.5, 1.5)) { Tuple2.apply }
+
+  property("ursem3") = forAll(genUrsem3) { g =>
+    ursem3(g) >= -3.0
+  } && ursem3((0.0, 0.0)) === -3.0
+
+  property("ursem4") = forAll(gen2(-2.0, 2.0)) { g =>
+    ursem4(g) >= -1.5
+  } && ursem4((0.0, 0.0)) === -1.5
+
+  val genUrsemWaves =
+    (Gen.choose(-0.9, 1.2) |@| Gen.choose(-1.2, 1.2)) { Tuple2.apply }
+
```

Comment:
All these `genUrsm*` should either be abstracted in some way to be used from other places or we should perhaps have them inline in the property definition?